### DISCUSSION TITLE
reflashing ECU possible?

### DISCUSSION THREAD

**Discussion by DavidIgor** [cedricp/ddt4all#1000] [Q&A]
I have weird issue with TB control and no scan tool can diagnose it - there are no fault codes at all. So is there a way to reflash ECU to factory settings ?

**Comment by matzel82** [comment]
Which car, which problem ? Normally you can’t reset without Original Renault Tool. You need access to download the latest configuration file or software.

**Reply by DavidIgor** [reply]
Dacia Dokker H4M 2018 , TB is opening valve only for first 30% (in closed loop) then it just jumps to WOT with fully pressed APP and loop jumps to open!

**Comment by DavidIgor** [comment]
set parameters to default ... somehow

**Comment by matzel82** [comment]
Okay…try to Reset Engine Adaption

**Reply by DavidIgor** [reply]
that sounds like something i could do.... any instructions to follow?

**Reply by DavidIgor** [reply]
ok so engine adaptation .. I can reset with puren and elm327 usb tool right?

**Comment by DavidIgor** [comment]
@matzel82 is it same as when you disconnect battery for 30minutes? or Ill get more ? any hints on how to do so ?

**Reply by matzel82** [reply]
No..that’s called powerlatch and reboot.

**Reply by DavidIgor** [reply]
![image](https://github.com/cedricp/ddt4all/assets/122531259/d36ded13-df10-44ec-9952-d03d578544d5) this is what I should do? but which one, or how many I can reset?

**Comment by KarelSvo** [comment]
Pyren menu RZ in the injection control unit.
You still haven't shown the contents of the ID menu.

**Comment by DavidIgor** [comment]
perfect! thx... ID menu means ?

**Comment by KarelSvo** [comment]
You simply have to select a control unit found by scanning in Pyren to see the submenus.

https://youtu.be/h5IOZ-skiD8?t=342

**Comment by DavidIgor** [comment]
![image](https://github.com/cedricp/ddt4all/assets/122531259/5ebd19fa-c84c-4688-933f-18fb567c3a6c)![image](https://github.com/cedricp/ddt4all/assets/122531259/51c6c8c9-8398-447f-871e-28beb56278b4)

what can I reset? all?

**Reply by matzel82** [reply]
No please !!

Only RZ 001, RZ033, RZ003, RZ062, RZ131, RZ042.

attention ⚠️ use with risk. At clip this orders use with switch off Ignition an wait every time 9 minutes after use. Please be careful

**Reply by DavidIgor** [reply]
@KarelSvo told me to go RZ and now what? Im on hold for now ... please guys ...

**Reply by DavidIgor** [reply]
I want to reset what ever I can ... and with pyren and elm327 usb tool. so can you confirm I can go ahead and reset those you mentioned in previous message?

**Reply by matzel82** [reply]
In principle, you can reset everything. But it should not because in the worst case it deletes all values of the adaptation. What problem do you have anyway? What kind of problems does he cause? Maybe the solution is easier than you think.

**Reply by KarelSvo** [reply]
I'm still waiting for the copy & paste of the ID menu
and the exact error description - when the engine is running
What does the engine do that it shouldn't do?

**Reply by DavidIgor** [reply]
engine doesn't control throttle valve in full range = low on power  copy and paste is difficult as Im using pc that has no internet

**Reply by matzel82** [reply]
Than RZ131 is only the one.

Start RZ, than switch of the ignition and wait lost connection or 9 minutes. If this doesn't work you need a new one. Have you made a test to check the Potentiometers at the Throttel ?

**Reply by DavidIgor** [reply]
@matzel82 engine adaptations what harm does it make- if I reset ?

**Reply by DavidIgor** [reply]
@matzel82 sorry ... i need a new one ? what ?

**Reply by matzel82** [reply]
> @matzel82 sorry ... i need a new one ? what ? 

Throttle Valve

**Reply by matzel82** [reply]
> @matzel82 engine adaptations what harm does it make- if I reset ? 
> 

It is possible that the engine will then run very badly for some time. You reset the learning values and adaptations, so to speak. You don't do anything to do anything. I just wouldn't necessarily recommend it. Then only reset the necessary.

**Reply by DavidIgor** [reply]
@matzel82 I have done only number 1 and 13 as you fellas scared me so bad! 
I hit reset on number 1 and turned off ignition then after about 20 minutes I reset the number 13 and turned off ignition . 
haven't testdrived yet.

**Reply by DavidIgor** [reply]
![image](https://github.com/cedricp/ddt4all/assets/122531259/967c49dc-7d84-458c-afa8-aebdbab60252) what are those "warn" values?

**Reply by KarelSvo** [reply]
Give me the dumps folder and savedEcus file.

**Reply by KarelSvo** [reply]
What is your native language?

**Reply by DavidIgor** [reply]
Slovakian language.. and here are the files : 
[Dacia Dokker H4M 2018 .zip](https://github.com/cedricp/ddt4all/files/13927926/Dacia.Dokker.H4M.2018.zip)

**Reply by KarelSvo** [reply]
There is a software update for your engine to combat speed problems.
Your engine control unit shows 237100546S.
This is not a current version.
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237105468R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237615468R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237106397R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237616397R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237107403R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237617403R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237108760R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237618760R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237109902R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237619902R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237103184S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237613184S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237100546S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237610546S/NON/INJ 
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237101811S/237103766S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237101811S/237613766S/OUI/INJ
The last number is ID001 in the ID menu, the calibration number.
In front of it is the code for the firmware.
If this code changes, a fundamental error has been fixed.

Put [this file](https://transfiles.ru/xwtca) in the doc folder.
If you are in the DE menu and have selected an error code, you can press the h key and the manual will open for this error code.
Did you see that Czech is available in the PyRen language selection?

**Reply by DavidIgor** [reply]
OMG .... ok ....  is it possible to update software with pyren or ddt4all?  what do I need to do and get to be able to do so? 
or is it the job for authorized Dacia service ? ( this should be a recall or something ....). 

btw:
i dont need czech language in pyren.

**Reply by matzel82** [reply]
> There is a software update for your engine to combat speed problems.
> 
> Your engine control unit shows 237100546S.
> 
> This is not a current version.
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237105468R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237615468R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237106397R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237616397R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237107403R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237617403R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237108760R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237618760R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237109902R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237619902R/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237103184S/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237613184S/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237100546S/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237610546S/NON/INJ 
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237101811S/237103766S/NON/INJ
> 
> K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237101811S/237613766S/OUI/INJ
> 
> The last number is ID001 in the ID menu, the calibration number.
> 
> In front of it is the code for the firmware.
> 
> If this code changes, a fundamental error has been fixed.
> 
> 
> 
> Put [this file](https://transfiles.ru/xwtca) in the doc folder.
> 
> If you are in the DE menu and have selected an error code, you can press the h key and the manual will open for this error code.
> 
> Did you see that Czech is available in the PyRen language selection?
> 
> 

How and where you have found this file and information ??

**Reply by DavidIgor** [reply]
I have managed to clear all DTCs (most of them where caused by me disconnecting and measuring things while I was trying to diagnose my throttle body issue). Also I successfully relearned Throttle valve positions (I will take it for a spin and report back)! 
Regarding the Outdated ECU firmware- I have not been able to find any information online... I hope that you guys can guide me with this FW update if it can be done at home ?!

@KarelSvo I even found "Slovenscinu" Slovak language in pyren - but translation is incorrect! it seems like mix of Polish and Czech languages.(Czech and Slovak languages are similar enough! so maybe just remove Slovak language). 

Thank you for your help! 
![image](https://github.com/cedricp/ddt4all/assets/122531259/2c240798-7ab4-4a4b-9a19-70a856ec7d2b)

**Reply by KarelSvo** [reply]
Slovenscinu = [SK]slovensky = [EN]Slovenian

**Reply by DavidIgor** [reply]
Slovenian is not Slovakian language ! 
Slovenscina = [SK]Slovensky = [EN]Slovak
Slovinscina = [SL]Slovinsky = [EN]Slovenian

hope it helps 
(again.... Slovak and Czech languages are similar enough that people can understand each other)!

**Reply by KarelSvo** [reply]
There is just no Slovak language file, only a Slovenian file.
The names of the language files are written in the local language.
That's why your conclusion doesn't fit.

>I even found "Slovenscinu" Slovak language in pyren

**Reply by DavidIgor** [reply]
this suppose to be Slovak language! 
![image](https://github.com/cedricp/ddt4all/assets/122531259/0208e3eb-c4d3-4984-8bda-f966259106c2)

**Reply by KarelSvo** [reply]
This is 'SLOVEN**Š**ČINA' without special characters, because of ASCII encoding of Python 2.
Not 'SLOVENČINA'

**Reply by DavidIgor** [reply]
copy that!

**Reply by DavidIgor** [reply]
@KarelSvo what about that ECU FW update?  how can I solve it ?

**Reply by AndyIreland** [reply]
FW can only be updated for ECU's  at Renault garage I reckon, using CAN CLIP and connection to Renault servers

**Reply by DavidIgor** [reply]
@AndyIreland Ill call dacia service tomorrow!

**Reply by AndyIreland** [reply]
ah yes thats right you have a Dacia , I dont know why I keep thinking you have a Renault - Mind you , Renault & Dacia are part of the same companies these days . - You most probably find that you could even leave your Dacia into a Renault garage and that they could even update the Firmware too

**Reply by DavidIgor** [reply]
yes I knew that, I hope they will take it as some recall task ... otherwise I have no idea How much it will cost me.

**Reply by DavidIgor** [reply]
@AndyIreland sorry to bother you... resetting adaptations lets say for ignition coils - is task to do after replacing new ignition coils? same for injectors, and rest of the options?

**Reply by KarelSvo** [reply]
Didn't read the diagnostic manual?

**Reply by AndyIreland** [reply]
I think only Safety concerns/issues like to do with braking etc would be a recall I dont think for ECU software but I could be wrong - I though as part of servicing (like leaving car in for its oil change etc) with the main dealer, they are supposed to check the ECU firmwares and update them accordingly . 

i dont know answer about the ignistion coils sorry . 

Sometimes on car's if you leave the 12v battery disconnected for a long time (say overnight) it drains down ECU's and they have to 're-learn' all the parameters of driving habits etc as you start driving again.

**Reply by DavidIgor** [reply]
@KarelSvo diagnostic manual? source?

**Reply by matzel82** [reply]
> @AndyIreland sorry to bother you... resetting adaptations lets say for ignition coils - is task to do after replacing new ignition coils? same for injectors, and rest of the options? 

You should only reset the adaptions for this parts you have changed. I you have a mechanical problem and repair the engine, so you can reset a lot of adaptations to "restarting" adapting at the ECU. (Lambda, Ignition, Flyweel etc.)

**Reply by KarelSvo** [reply]
@DavidIgor The file you should put in the doc folder.

**Reply by DavidIgor** [reply]
@KarelSvo that is written by google translate... it is mostly unusable. 
do you have same in English?( written by human)?

**Reply by KarelSvo** [reply]
The original Renault, every Czech and Slovakian diagnostic technician reads this on his device.

t.me/pyren_bot
send
/doc   yourvinnumber  GB

**Reply by DavidIgor** [reply]
@KarelSvo  I don't have telegram ..

**Reply by DavidIgor** [reply]
@KarelSvo  hope you doing well ! 
I have contacted Dacia/Renault service and they told me that update is not covered nor part of any recall ( at least not for my type of car ).  They also told me that update will take anywhere form 30-60 minutes and they do not think that I will feel any difference. 

Im going to book it ... so you found that my current ECU version is : 237100546S
and it should be one of those (after update)? :
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237105468R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237615468R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237106397R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106313R/237616397R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237107403R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237617403R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237108760R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237618760R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237109902R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237619902R/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237103184S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237613184S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237100546S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237106933R/237610546S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237101811S/237103766S/NON/INJ
K67/AV/H4M/738//E06R/////0000CRQ72A/237105698R/237101811S/237613766S/OUI/INJ


note: 
I will double check version after - I hope that i will get report that will state installed version of FW. (I wont be showing these versions to them prior). 

Thank You

**Comment by AndyIreland** [comment]
> ![image](https://private-user-images.githubusercontent.com/122531259/296447626-967c49dc-7d84-458c-afa8-aebdbab60252.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUxNTA5MDEsIm5iZiI6MTcwNTE1MDYwMSwicGF0aCI6Ii8xMjI1MzEyNTkvMjk2NDQ3NjI2LTk2N2M0OWRjLTdkODQtNDU4Yy1hZmE4LWFlYmRiYWI2MDI1Mi5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTNUMTI1NjQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmE5NmM1Yjg5ZmVjMTc0MmZiNTFkZmE3NDk1OTI5OWY2ZWQ1MDhhM2VlYWU5ZWNjYTA4Y2JjOTc2YTYyMTI3NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Gfvr5LGPR612lee0qhfA03dHZL0eC2gmDF8kulV8Cng) what are those "warn" values?

memorised and present DTC error codes found in the ECU's

**Comment by DavidIgor** [comment]
@AndyIreland weird as I can not see any DTCs while scanning with ddt4all or pyren ....  how to clear it all ? is there a way to clear all DTCs  ? 
note: 
Injection has 0  so my reset did worked ...well at least on that one

**Reply by KarelSvo** [reply]
The absolute number of errors is shown under Warn.
Only the errors valid for this ECU version are displayed in the DE menu.
According to your dumps, you have only opened the submenus of 2 Ecus.

**Comment by DavidIgor** [comment]
Im looking to verify version of my ECU software,

**Reply by AndyIreland** [reply]
yep ,you can do that through PyREN too - and DDT4ALL

**Reply by DavidIgor** [reply]
ok but how where? I checked dumps from pyren but nothing to see there

**Reply by AndyIreland** [reply]
well say now for example i want to see what version my TCU (telecommunications control unit) software is installed - I bring this up in PyREN:
![image](https://github.com/cedricp/ddt4all/assets/49172217/e4224ca4-a62d-4fbc-97c2-e02670756d8f)

Then i select 6.)
![image](https://github.com/cedricp/ddt4all/assets/49172217/9f5244e8-4854-4532-9ccc-160ee2356a9d)

Then I select 2.)

![image](https://github.com/cedricp/ddt4all/assets/49172217/0d015985-5ab2-4dc6-a6b4-0a794f4d23a8)



And so the software version of my TCU on my car is 0504

**Comment by KarelSvo** [comment]
Still in the ID menu, which I've mentioned several times.

**Comment by DavidIgor** [comment]
so I will go "1" injection?

**Reply by KarelSvo** [reply]
Don't you dare select a control device from the list of control devices found during scanning?

**Reply by DavidIgor** [reply]
haha, so software versions is actual version? and electronic version is latest available? Do I understand it the right way?

**Reply by DavidIgor** [reply]
![image](https://github.com/cedricp/ddt4all/assets/122531259/a23b2f68-f9fa-4501-bce5-c88528e470bb)

**Comment by DavidIgor** [comment]
@AndyIreland this is what I got: 
![image](https://github.com/cedricp/ddt4all/assets/122531259/4a4393b7-20fe-4a3d-8b55-af75e74ce2af)
I guess something will change after they in Dacia service update the ECU? !

**Reply by AndyIreland** [reply]
so your installed software version listed on your ecu at present is **5300** - if they have an update and it cures some problems they might update it for you if you ask them nicely. - if its not at service time or a recall you may have to pay a diagnostic fee + fee for updating the software

**Comment by DavidIgor** [comment]
all right, it slowly becoming more clear.. ID007 is number written on ECU cover ( part number ). ID005 is actual version you saying, but how did @KarelSvo know that it is outdated !? and he listed like 16 lines of other ECUs? but nothing was showing anything close to 5300 (ID005)...
-sorry for being pain in the ass.

**Reply by AndyIreland** [reply]
oh right - maybe I am the one incorrect then . i would have thought logically that 5300 is the installed version and that is the one you have to check with Dacia if there is a newer version or not ... but i am no expert or technician, I am just an enthusiast

**Reply by DavidIgor** [reply]
well, I don't know.... that's certain ! 
After @KarelSvo mentioned that my ECU has old software I did research and indeed I found that other people had similar issues with throttle body and rpm/power issues linked to outdated software.

**Comment by KarelSvo** [comment]
ID007 is the hardware identification number. It doesn't change in the list. The update module only cares about ID001. The list shows
technical code/hardware ID007/firmware/calibration ID001
Same technical code means that all software versions belong to the same vehicle type. OUI is the most current version at the time the list was created.

**Comment by DavidIgor** [comment]
@KarelSvo copy that ! 
so my current ID001 is 237100546S and latest available (OUI) at the time you provided that list was/is 237613766S 
I can see my ID001 is 3 updates old ! 
And can you please explain that fundamental fix  if  certain numbers are different .... I guess numbers in ID001 right?

**Comment by DavidIgor** [comment]
@KarelSvo 
8 was in renault today and they did this: 
![image](https://github.com/cedricp/ddt4all/assets/122531259/bb9ba03f-710a-439b-bb71-c13891cf591a)

no numbers match your list.

**Comment by AndyIreland** [comment]
so software version changed from 5300 to 5c00 - How much did they charge to update the software if you dont mind me asking , dont worry if you dont want to answer and sorry if I am being cheeky asking

**Comment by DavidIgor** [comment]
@AndyIreland bit under 80e and it took ca 90minutes.

**Reply by AndyIreland** [reply]
Thanks, I think personally thats not too bad a price to update the software on your ECU - they easily charge 150+ sometimes just to diagnose car problems.  - the 90minutes must have been because it was a large file to download I would say? - in its Gb's in size possibly

**Reply by DavidIgor** [reply]
absolutely! im happy. car does feel bit better to drive as well, will see...

**Reply by DavidIgor** [reply]
@AndyIreland I need to fix me new tpms sensors they are transmitting but car won't read them.... do I have to manually type sensors ID in DDT4ALL?  screenshot is with old tpms sensors ![image](https://github.com/cedricp/ddt4all/assets/122531259/203752c8-57be-4cee-9ed6-28ab1aa8161c)

when I hit "TP set" on dash... tpms warning light disapear but only for like 10-15km and it appears back. new sensors are transmitting as I checked each and grabbed IDs.

**Reply by AndyIreland** [reply]
Im sorry David I cannot help you on that one, I dont know how the TPMS system works andI have not even got TPMS on my car (old Renault Zoe EV) 
Hope someone else can answer your question
